backend/src/main/java/de/kathahuth/gymast/service/UserService.java

```diff
@@ -0,0 +1,29 @@
+package de.kathahuth.gymast.service;
+
+import de.kathahuth.gymast.db.UserMongoDb;
+import de.kathahuth.gymast.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserService {
+
+    private final UserMongoDb userMongoDb;
+
+    @Autowired
+    public UserService(UserMongoDb userMongoDb) {
+        this.userMongoDb = userMongoDb;
+    }
+
+    public List<User> listUsers(){
+        return userMongoDb.findAll();
+    }
+
+    public Optional<User> getUserByUsername(String username) {
+        return userMongoDb.findById(username);
```

Comment:
you might need a null check here in case the user is not present... but I'm just not sure if it's covered by the exception handling in the controller ^^'
and your test does show that it returns "empty" for an unknown user... how does that work? ^^

Reply:
This is perfectly fine, returning an empty Optional is standard for `findBy` methods of mongodb.